src/main/webapp/js/chart-loader.js

```diff
@@ -79,13 +82,27 @@
     chart.draw(diffFavoriteTea, pieOptions);
   };
 
+  function fetchMessageData() {
+    const url = "/messagechart";
+    if(_messageChartPromise == null) {
+      _messageChartPromise = fetch(url)
+      .then((response) => { 
+        return response.json();
+      })
+      .then((msgJson) => {
+        console.log(msgJson);
```

Comment:
Also add:

     return msgJson

So that the object is passed along to whatever function is registered with calling `.then` on the result of `fetchMessageData()`.